ompi/mca/coll/han/coll_han_alltoallv.c

```diff
@@ -194,31 +270,43 @@ static inline int alltoallv_sendrecv_w_direct_for_debugging(
                 jrecvs_completed++;
         }
 
-        requests[jreq] = &ompi_request_null.request;
         if (ii_send_req && jsends_posted < ntypes_send) {
             rc = ompi_datatype_create_contiguous( 1, (ompi_datatype_t*)send_types[jsends_posted], &yuck_ompi_dtype_from_opal );
-            ompi_datatype_commit(&yuck_ompi_dtype_from_opal);
-            MCA_PML_CALL(isend
+            if (rc) break;
+            rc = ompi_datatype_commit(&yuck_ompi_dtype_from_opal);
+            if (rc) break;
+            rc = MCA_PML_CALL(isend
                 (send_from_addrs[jsends_posted], (int)send_counts[jsends_posted], yuck_ompi_dtype_from_opal, jrank_sendto,
                 MCA_COLL_BASE_TAG_ALLTOALLV, MCA_PML_BASE_SEND_STANDARD,
                 comm, &requests[jreq]));
-            ompi_datatype_destroy( &yuck_ompi_dtype_from_opal );
+            if (rc) break;
+            rc = ompi_datatype_destroy( &yuck_ompi_dtype_from_opal );
+            if (rc) break;
+
             jsends_posted++;
         }
         if (!ii_send_req && jrecvs_posted < ntypes_recv ) {
             rc = ompi_datatype_create_contiguous( 1, (ompi_datatype_t*)recv_types[jrecvs_posted], &yuck_ompi_dtype_from_opal );
-            ompi_datatype_commit(&yuck_ompi_dtype_from_opal);
-            MCA_PML_CALL(irecv
+            if (rc) break;
+            rc = ompi_datatype_commit(&yuck_ompi_dtype_from_opal);
+            if (rc) break;
+            rc = MCA_PML_CALL(irecv
                 (recv_to_addrs[jrecvs_posted], (int)recv_counts[jrecvs_posted], yuck_ompi_dtype_from_opal, jrank_recvfrom,
                 MCA_COLL_BASE_TAG_ALLTOALLV,
                 comm, &requests[jreq]));
-            ompi_datatype_destroy( &yuck_ompi_dtype_from_opal );
+            if (rc) break;
+            rc = ompi_datatype_destroy( &yuck_ompi_dtype_from_opal );
+            if (rc) break;
             jrecvs_posted++;
         }
 
-
         if (rc) { break; };
     }
+    if (rc) {
+        opal_output_verbose(1, mca_coll_han_component.han_output,
+            "Failed in alltoallv_sendrecv_w_direct_for_debugging: jloop=%d, rc=%d\n",
```

Comment:
what debugging ?

Reply:
That function is inefficient and not used.  However if you doubt that something is correct in all the business logic of the main sendrecv_w, then it is a easy code change to use this function to double-check.

Is there a better way to keep such a function but also keep it out of the way?